.github/workflows/reusable-solidity-docs.yml

```diff
@@ -0,0 +1,248 @@
+name: Generate and push Solidity API documentation
+
+# To use this workflow, make sure `solidity-docgen` package is added to the 
+# projects's dev dependencies and is configured in Hardhat config (see
+# https://github.com/OpenZeppelin/solidity-docgen#usage).
+# The workflow expects `outputDir` to be set to `generated-docs`. You may
+# also need to specify other configs, like `temlates` or `exclude`.
+on:
+  workflow_call:
+    inputs:
+      projectSubfolder:
+        description: "A root's subfolder containing the `contracts` folder and
+          the Yarn / Hardhat configuration. Leave empty ('') if the project
+          config resides in the root. If config resides in a subfolder, provide
+          the subfolder name/path, without leading `.` and ending `/`. For
+          example, type `/v2/solidity` if the path to the `contracts` folder is
+          `./v2/solidity/contracts`."
+        type: string
+        required: false
+        default: ""
+      removeTrailingSpacesInComments:
+        description: "True if you want to remove space characters between `///`
+          and a comment text in the Solidity files, before generating `.md`
+          files. This may be needed to improve the look of the lists in the
+          generated files."
+        type: boolean
+        required: false
+        default: true
+      preProcessingCommand:
+        description: "An optional additional bash command to be executed before
+          transforming the Solidity files to an HTML file. The command will be
+          executed in the path specified by `projectSubfolder`."
+        type: string
+        required: false
+      addTOC:
+        description: "True if you want to add Table Of Contents to the generated
+          Markdown file. Uses `markdown-toc` tool to generate TOC. Shouldn't be
+          set to `true` if Docgen's `pages` attribute in `hardhat.config.ts` is
+          set to `items` or `files`."
+        type: boolean
+        required: false
+        default: true
+      tocOptions:
+        description: "Options to be passed to the `markdown-toc` tool used for
+          generation of Table Of Contents. See
+          https://github.com/jonschlinkert/markdown-toc#cli for more details."
+        type: string
+        required: false
+        default: "-i --maxdepth 2"
+      publish:
+        description: "True if you want to push the generated file(s) to the
+          destination repository."
+        type: boolean
+        required: false
+        default: true
+      verifyCommits:
+        description: "True if you want to sign commits updating the docs using
+          GPG key. You'll need to specify XXX and YYY secrets."
```

Comment:
Are `XXX` and `YYY` placeholders that should be replaced with `gpgPrivateKey` and `gpgPassphrase`?

Reply:
Yes, I was supposed to update that 🤦‍♀️. Thanks for pointing this out!

Reply:
See 283d7a77c0b64b1978f6e6aae3f0a8d1e6553a7b.